Orbitersdk/samples/ProjectApollo/src_csm/saturnmesh.cpp

```diff
@@ -271,6 +271,105 @@ void CMLETHoriCoeffFunc(double aoa, double M, double Re, double *cl, double *cm,
 	*cm = factor * (frac*CM[j + 1] + (1.0 - frac)*CM[j]);
 }
 
+void CSMAeroVertCoeff(VESSEL *v, double aoa, double M, double Re, void *context, double *cl, double *cm, double *cd)
+{
+	//Use this definition of the aoa instead
+	VECTOR3 vec;
+	v->GetAirspeedVector(FRAME_LOCAL, vec);
+	double aoa_T = acos(unit(vec).z);
+
+	double Kn = M / Re * 1.482941286; //Knudsen number. Factor is sqrt(1.4*pi/2)
+	int i, j;
+	const int nabsc = 9;
+	static const double AOA[nabsc] = { 0 * RAD, 10 * RAD, 20 * RAD, 30 * RAD, 90 * RAD, 150 * RAD, 160 * RAD, 170 * RAD, 180 * RAD };
+	static const double CD_free[nabsc] = { 2.41, 2.73, 3.14, 3.61, 5.01, 2.92, 2.46, 2.08, 1.96}; //free flow
+	static const double CD_cont[nabsc] = { 0.54, 0.59, 0.77, 1.12, 2.78, 1.32, 1.41, 1.50, 1.69 }; //continuum flow
+	static const double CM_free[nabsc] = { 0, 0.0440, -0.1436, -0.4349, -0.7682, -0.2369, 0.3820, -0.0115, 0 };
+	static const double CM_cont[nabsc] = { 0, 0.0977, 0.1069, 0.0230, -0.5803, 0.0206, 0.0981, 0.0145, 0 };
+
+	//For drag
+	for (i = 0; i < nabsc - 1 && AOA[i + 1] < aoa_T; i++);
```

Comment:
It took me a while to understand what you are doing here. While a loop without body is totally valid code, it does increase the cognitive load of the person reading it as it is not obvious that you use i later on.

To make it clear you are using this value how about doing it in the loop body? Something like:
```c
int someVar;
for (int i = 0; i < x ; i++)
    someVar++;
```
The compiler will most likely optimize it away (if you are worried about that) nor will it affect complexity.
Alternatively, I would at the very least comment what you are doing and change i to a descriptive name.